index/spec_index.go

```diff
@@ -1698,18 +1701,93 @@ func (index *SpecIndex) extractDefinitionsAndSchemas(schemasNode *yaml.Node, pat
 			name = schema.Value
 			continue
 		}
+
 		def := fmt.Sprintf("%s%s", pathPrefix, name)
 		ref := &Reference{
-			Definition: def,
-			Name:       name,
-			Node:       schema,
-			Path:       fmt.Sprintf("$.components.schemas.%s", name),
-			ParentNode: schemasNode,
+			Definition:            def,
+			Name:                  name,
+			Node:                  schema,
+			Path:                  fmt.Sprintf("$.components.schemas.%s", name),
+			ParentNode:            schemasNode,
+			RequiredRefProperties: index.extractDefinitionRequiredRefProperties(schemasNode, map[string][]string{}),
 		}
 		index.allSchemas[def] = ref
 	}
 }
 
+// extractDefinitionRequiredRefProperties goes through the direct properties of a schema and extracts the map of required definitions from within it
+func (index *SpecIndex) extractDefinitionRequiredRefProperties(schemaNode *yaml.Node, reqRefProps map[string][]string) map[string][]string {
+	if schemaNode == nil {
+		return reqRefProps
+	}
+
+	_, requiredSeqNode := utils.FindKeyNode("required", schemaNode.Content)
```

Comment:
I would use `utils.FindKeyNodeTop` instead of `utils.FindKeyNode`. The latter will look through two levels for the key, vs just the top. It can give inaccurate results. and find child keys.

Reply:
Cool, I've updated the PR! :)